external-crates/move/crates/move-stackless-bytecode-2/src/stackless/translate.rs

```diff
@@ -0,0 +1,602 @@
+
+        IB::LdFalse => assign_reg!([push!()] = imm!(Value::Bool(false))),
+
+        IB::CopyLoc(loc) => {
+            assign_reg!(
+                [push!()] = RValue::Local {
+                    op: LocOp::Copy,
+                    arg: *loc as usize
+                }
+            )
+        }
+
+        IB::MoveLoc(loc) => {
+            assign_reg!(
+                [push!()] = RValue::Local {
+                    op: LocOp::Move,
+                    arg: *loc as usize
+                }
+            )
+        }
+
+        IB::StLoc(loc) => Instruction::StoreLoc {
+            loc: *loc as usize,
+            value: Register(pop!()),
+        },
+
+        IB::Call(function_ref) => {
+            let name = &function_ref.module.name;
+            let mut modules = ctxt.model.modules();
+            let module = modules
+                .find(|m| {
+                    m.compiled().name() == (&function_ref.module.name)
+                        && *m.compiled().address() == function_ref.module.address
+                })
+                .unwrap_or_else(|| {
+                    panic!(
+                        "Module {} with address {} not found in the model",
+                        name, function_ref.module.address
+                    )
+                });
+            let compiled = module.compiled();
+            let function = compiled
+                .functions
+                .get(&function_ref.function)
+                .unwrap_or_else(|| {
+                    panic!(
+                        "Function {} not found in module {}",
+                        function_ref.function, name
+                    )
+                });
+
+            let args = make_vec!(function.parameters.len(), Register(pop!()));
+
+            let lhs = make_vec!(function.return_.len(), push!());
+
+            Instruction::AssignReg {
+                lhs,
+                rhs: RValue::Call {
+                    function: function.name,
+                    args,
+                },
+            }
+        }
+
+        IB::Pack(struct_ref) => {
+            let args = make_vec!(struct_ref.struct_.fields.0.len(), Register(pop!()));
+            assign_reg!(
+                [push!()] = RValue::Data {
+                    op: DataOp::Pack(struct_ref.clone()),
+                    args
+                }
+            )
+        }
+
+        IB::Unpack(struct_ref) => {
+            let rhs = RValue::Data {
+                op: DataOp::Unpack(struct_ref.clone()),
+                args: vec![Register(pop!())],
+            };
+            let lhs = make_vec!(struct_ref.struct_.fields.0.len(), push!());
+            Instruction::AssignReg { rhs, lhs }
+        }
+
+        IB::ReadRef => assign_reg!([push!()] = data_op!(DataOp::ReadRef, Register(pop!()))),
+
+        IB::WriteRef => {
+            assign_reg!([] = data_op!(DataOp::WriteRef, Register(pop!()), Register(pop!())))
+        }
```

Comment:
Consider making this a command instead of a data operation that returns no values.